pkg/runtime/core/trainingruntime.go

```diff
@@ -103,26 +108,74 @@ func (r *TrainingRuntime) buildObjects(
 		runtime.WithAnnotations(propagationAnnotations),
 		runtime.WithMLPolicy(mlPolicy),
 		runtime.WithPodGroupPolicy(podGroupPolicy),
+		runtime.WithTemplateSpecObjApply[jobsetv1alpha2ac.JobSetSpecApplyConfiguration](&jobsetv1alpha2.JobSet{
```

Comment:
I wonder if loading the TrainingRuntime as apply configuration directly would simplify things and remove the need for converting things like Volume to VolumeApplyConfiguration.

Reply:
Yeah, that is a good point. Actually, I was thinking about it a bit. 
However, we will add another CRD as runtime like SingleRoleTrainingRuntime based on batch/v1 Job.

Reply:
Ah, does that mean we can pass PodSpec as ApplyConfiguration to WithTemplateSpecObjApply, right?

Reply:
In that case, we can reduce parse from `WithTemplateSpecObjApply` function. That makes sense. Thanks.

Reply:
>However, we will add another CRD as runtime like SingleRoleTrainingRuntime based on batch/v1 Job.

I am not sure if that is needed, until we really hear use-cases when JobSet won't work for users.

Reply:
> > However, we will add another CRD as runtime like SingleRoleTrainingRuntime based on batch/v1 Job.
> 
> I am not sure if that is needed, until we really hear use-cases when JobSet won't work for users.

As I mentioned above, the unstructured parse mechanism will be removed. The SingleRoleTrainingRuntime was just example.

Reply:
Done.